coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -2715,39 +2752,32 @@ def _parse_keyword_args(context, node, weight, bias) -> Tuple[Var]:
     x, num_groups, weight, bias, eps = _parse_positional_args(context, node)
     weight, bias = _parse_keyword_args(context, node, weight, bias)
 
-    n,c = x.shape[0],x.shape[1] # at minimum (N, C) required
-    num_groups = builtins.min(num_groups,c)
-    new_shape = [n, num_groups, c//num_groups]
-    # optimization for non symbolic shapes. This get rids of 3 mil ops that required on dynamic shapes
-    if not any_symbolic(x.shape[2:]):
-        new_shape += [*x.shape[2:]] # adds remaining dims
-        input_shape = [*x.shape] # n, c, *
-    else:
-        input_shape = mb.shape(x=x)
-        input_shape_sliced = mb.slice_by_size(x=input_shape, begin=[2], size=[-1]) # x_shape[2:]
-        new_shape = mb.concat(values=[new_shape, input_shape_sliced], axis=0)
+    result = _group_norm_impl(x, num_groups, weight, bias, eps)
+    context.add(result, node.name)
 
-    num_extra_axes = len(x.shape[2:])
-    axes_ = [int(i) for i in range(2, 2 + num_extra_axes + 1)]
-    weight_shape, bias_shape = [1,c], [1,c]
-    weight_shape += [1 for _ in range(num_extra_axes)]
-    bias_shape += [1 for _ in range(num_extra_axes)]
 
-    x = mb.reshape(x=x, shape=new_shape)
-    mean = mb.reduce_mean(x=x, axes=axes_, keep_dims=True)
-    var = _var(x, axes=axes_, keep_dims=True, unbiased=False)
-    var_plus_eps = mb.add(x=var, y=eps)
-    std = mb.sqrt(x=var_plus_eps)
-    x = mb.sub(x=x, y=mean)
-    x = mb.real_div(x=x, y=std)
-    x = mb.reshape(x=x, shape=input_shape)
-    if weight is not None:
-        weight = mb.reshape(x=weight, shape=weight_shape)
-        x = mb.mul(x=x,y=weight)
-    if bias is not None:
-        bias = mb.reshape(x=bias, shape=bias_shape)
-        x = mb.add(x=x, y=bias)
-    context.add(x,node.name)
+@register_torch_op
+def native_group_norm(context, node):
+    def _parse_positional_args(context, node) -> Tuple[Var]:
+        inputs = _get_inputs(context, node, expected=8)
+
+        x = inputs[0]
+        weight = inputs[1]
+        bias = inputs[2]
+        N = inputs[3].val
```

Comment:
Better/longer variable names would be good here. Looking at the [PyTorch docs](https://pytorch.org/docs/stable/generated/torch.nn.GroupNorm.html#torch.nn.GroupNorm), I'm not really sure what some of there variable are, i.e. `groups`, `N`, `C`. Having them match the PyTorch docs, as much as possible, would be great.

Reply:
Yeah this is a lower level op that is not in PyTorch doc 😮‍💨 It is defined in PyTorch native functions
```
- func: native_group_norm(Tensor input, Tensor? weight, Tensor? bias, SymInt N, SymInt C, SymInt HxW, int group, float eps) -> (Tensor, Tensor, Tensor)
  dispatch:
    CPU, CUDA: native_group_norm
    CompositeExplicitAutograd: math_group_norm
  autogen: native_group_norm.out
  tags: core
```